test/services/supplementary-billing/create-billing-invoice.service.test.js

```diff
@@ -0,0 +1,40 @@
+'use strict'
+
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+
+const { describe, it, beforeEach } = exports.lab = Lab.script()
+const { expect } = Code
+
+// Test helpers
+const BillingInvoiceModel = require('../../../app/models/water/billing-invoice.model.js')
+const DatabaseHelper = require('../../support/helpers/database.helper.js')
+const InvoiceAccountHelper = require('../../support/helpers/crm-v2/invoice-account.helper.js')
+
+// Thing under test
+const CreateBillingInvoiceService = require('../../../app/services/supplementary-billing/create-billing-invoice.service.js')
+
+describe('Create Billing Invoice service', () => {
+  const billingBatchId = '95177485-bfe7-48ed-b2f0-dba5a5084b45'
+  const billingPeriod = { startDate: new Date('2022-04-01'), endDate: new Date('2023-03-31') }
+
+  beforeEach(async () => {
+    await DatabaseHelper.clean()
+  })
+
+  it('returns the new billing invoice instance containing the correct data', async () => {
+    const invoiceAccount = await InvoiceAccountHelper.add({ invoiceAccountNumber: 'TEST12345A' })
+    const chargeVersion = { invoiceAccountId: invoiceAccount.invoiceAccountId }
```

Comment:
This is a personal preference thing. I use `beforeEach()` blocks to handle the 'setup' for a test. The aim is that what happens in the test, i.e. the `it` block, is limited to executing the thing and asserting the result or behaviour.

That preference has leached into our unit tests as a defacto convention, which is why I'm flagging this.

So, I'd suggest declaring `invoiceAccount` and `chargeVersion` where you are already declaring your other values and then instantiating them in the `beforeEach()` instead of in the test.

```javascript
describe('Create Billing Invoice service', () => {
  const billingBatchId = '95177485-bfe7-48ed-b2f0-dba5a5084b45'
  const billingPeriod = { startDate: new Date('2022-04-01'), endDate: new Date('2023-03-31') }

  let invoiceAccount
  let chargeVersion

  beforeEach(async () => {
    await DatabaseHelper.clean()

    invoiceAccount = await InvoiceAccountHelper.add({ invoiceAccountNumber: 'TEST12345A' })
    chargeVersion = { invoiceAccountId: invoiceAccount.invoiceAccountId }
  })

  // ...
```

Reply:
Sounds sensible, I shall implement your suggestion.